services/agents/connection_checker.go

```diff
@@ -100,6 +100,12 @@ func (c *ConnectionChecker) CheckConnectionToService(ctx context.Context, q *ref
 	}
 	l.Infof("CheckConnection response: %+v.", resp)
 
+	// we can get nil response if appropriate request is not implemented
+	// on the agent side (e.g. because of the old version of pmm-agent).
+	if resp == nil {
+		return errors.New("nil response")
+	}
+
```

Comment:
shouldn't it be handled in `SendAndWaitResponse`?

Reply:
I thought about it, but I found this comment  😄 
https://github.com/percona/pmm-managed/blob/main/services/agents/channel/channel.go#L157

also, pmm-agent sends response without payload in the default case:
https://github.com/percona/pmm-agent/blob/main/client/channel/channel.go#L298

I could handle this nil payload, but I would like to have a call to discuss this.